codesign/test/file_codesing_visitor_test.dart

```diff
@@ -2,13 +2,65 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io' as io;
 import 'package:codesign/codesign.dart' as cs;
 import 'package:file/file.dart';
 import 'package:file/memory.dart';
 import 'package:logging/logging.dart';
 import 'package:test/test.dart';
 import './src/fake_process_manager.dart';
 
+/// An override of [cs.visitEmbeddedZip].
+///
+/// After extracting a folder from a zip file, (a fake unzip)
+/// [visitEmbeddedZipTest] delegates the extracted folder to
+/// [visitDirectoryPrintln], instead of the real [cs.visitDirectory] fucntion.
+/// This override is used when we don't want to actually perform the
+/// zip/unzip operations, but would like to verify the extracted folder.
+Future<void> visitEmbeddedZipTest(FileSystemEntity file, String entitlementParentPath, Directory tempDir,
```

Comment:
Thanks for reviewing! umm I injected a `visitiEmbeddedZipTest` function into `codeSignVisitor` in test because `visitEmbeddedZip` contains some fake processes such as `zip` and `unzip`. For example, to extract `Flutter.zip` into the `Flutter` folder, If the `zip` process is real, a `Flutter` folder would be present at the file system location, and `visitDirectory` can follow up from the folder location. In tests, the process of `unzip` is fake, and therefore no Flutter folder will be generated, and a `visitDirectory` call cannot follow up. 

I might be wrong, I thought the concept `injection` applies to function such as `visitiEmbeddedZip`, which I made as a part of the `codesignVisitor` constuctor. It takes a fake version of `visitiEmbeddedZip`, instead of taking the real version of it. 

An alternative way of interpreting `injection`, as I could think of, is to make process related functions such as `zip`, `unzip`, `getFileType`, `notarize` injectable. But this approach would add a lot of parameters to the constructor, and would still require an override of such functions, such as `zipTest`, `unzipTest` etc., so that they become injectable in tests. 

It would be great if Casey could elaborate a bit more on the `injection` part (maybe it means create an override of all the process related function calls, instead of creating the overrides of visit-functions?) Thank you!

Reply:
update: removed injections.